### DISCUSSION TITLE
Glow / Outline / Opacity

### DISCUSSION THREAD

**Discussion by slhide** [pokepetter/ursina#658] [Ideas]
Hello !
Ursina is currently my favourite module and here are 3 of my suggestions for it to look even better.
1- Would it be possible to add outline to shapes, either on edges like the cube below (like Entity(border=...))
![image](https://github.com/pokepetter/ursina/assets/139326331/092fb75c-3b1c-4f64-8141-68af6f96f1d1)
or around the global figure like in games like this one and rotating with the model 
![image](https://github.com/pokepetter/ursina/assets/139326331/bc8c0460-c995-49cd-8f24-58bcb4c4910f)
2- Maybe adding a glowing / shining effect around models like stars : Entity(glow=...)
3 - I managed to export alpha channel of a GLB file, it works well on Blender or 3D Viewer but the model appears as blank on Ursina. Why not make it possible to import models with alpha, since we can already change it with Entity(alpha=...)?

I know all of this might seem hard to make but it would be awesome -- at least for me !
Still, you made a great job for that module!

**Comment by aac3master** [comment]
> or around the global figure like in games like this one and rotating with the model

Good idea. Also easy. Model should show only surfaces that not visible on original and scaled a bit (x0.5 or any other). Then just parent and color.
![cube](https://github.com/pokepetter/ursina/assets/96526713/0a3b8bbf-863a-49ae-b912-9d59f40ae12d)

> Original

![cube2](https://github.com/pokepetter/ursina/assets/96526713/4c838e44-62f8-4702-85ad-f6a63de53e52)

> Shows only surfaces that not visible on original

**Comment by useless-hyphen** [comment]
@slhide I can only give you some hints, your wishes you have to fulfill yourself.
to 1) For your "outlines" you have to use a shader with edge-detection. 
to 2) Shining and glowing are also shader-topics. And you have to use one of the ursina Light-classes.
to 3) Alpha and model are two separate topics. Alpha belongs to the color of your entity and the model determines the geometric body of the entity. When you load your model in ursina, you do not simultaneously load the color. The white color of your entity is the default color when no color was given.
